pipelines/azure-pipelines-template.yml

```diff
@@ -16,36 +16,44 @@ resources:
 stages:
 - stage: Test
   displayName: Test
-  pool:
-    vmImage: $(poolVmImage)
+  pool: $(agentPool)
 
   jobs: 
-    - job: ApiTest
-      displayName: API
-      steps:
-        - task: DotNetCoreCLI@2
-          displayName: Unit
-          inputs:
-            command: test
-            projects: tests/API/Unit/Unit.csproj
-            
-        - task: DotNetCoreCLI@2
-          displayName: Integration
-          inputs:
-            command: test
-            projects: tests/API/Integration/Integration.csproj  
+  - job: Unit
+    steps:
+    - task: UseDotNet@2
+      inputs:
+        version: 3.1.x
+
+    - task: DotNetCoreCLI@2
+      inputs:
+        command: test
+        projects: tests/API/Unit/Unit.csproj
+          
+  - job: Integration
+    steps:
+    - task: UseDotNet@2
+      inputs:
+        version: 3.1.x
+
+    - task: DotNetCoreCLI@2
+      inputs:
+        command: test
+        projects: tests/API/Integration/Integration.csproj  
 
 - stage: Package
   displayName: Package  
   dependsOn: Test
```

Comment:
Packaging is fast, but if we want to increase performance we could just do it as soon as any agent is available.

Reply:
I love this idea. Done.